include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -141,45 +145,38 @@ struct alignment_configurator
      * \tparam config_type The configuration for which to select the correct matrix policy.
-     * \tparam trait_types A template parameter pack with additional traits to augment the selected policy.
+     * \tparam score_t The type used for the score matrix.
+     * \tparam trace_t The type used for the trace matrix.
      */
-    template <typename config_type, typename score_allocator_t, typename trace_allocator_t>
+    template <typename config_type, typename score_t, typename trace_t>
     struct select_matrix_policy
     {
     private:
 
-        //!\brief Selects the correct alignment matrix policy based on the stored config types.
-        template <typename config_t>
-        static constexpr auto select() noexcept
+        //!\brief Selects the proper score matrix.
+        static constexpr auto select_score_matrix() noexcept
         {
-            // Check whether traceback was requested or not.
-            if constexpr (std::is_same_v<typename trace_allocator_t::value_type, ignore_t>)
-            {  // No traceback
-                if constexpr (config_t::template exists<align_cfg::band>())
-                    return deferred_crtp_base<banded_score_dp_matrix_policy, score_allocator_t>{};
-                else
-                    return deferred_crtp_base<unbanded_score_dp_matrix_policy, score_allocator_t>{};
-            }
+            // Check whether banded or not.
+            if constexpr (config_type::template exists<align_cfg::band>())
+                return alignment_score_matrix_one_column_banded<score_t>{};
             else
-            {  // requested traceback
-                if constexpr (config_t::template exists<align_cfg::band>())
-                {
-                    return deferred_crtp_base<banded_score_trace_dp_matrix_policy,
-                                              score_allocator_t,
-                                              trace_allocator_t>{};
-                }
-                else
-                {
-                    return deferred_crtp_base<unbanded_score_trace_dp_matrix_policy,
-                                              score_allocator_t,
-                                              trace_allocator_t>{};
-                }
-            }
+                return alignment_score_matrix_one_column<score_t>{};
+        }
+
+        //!\brief Selects the proper trace matrix.
+        static constexpr auto select_trace_matrix() noexcept
+        {
+            if constexpr (config_type::template exists<align_cfg::band>())
+                return alignment_trace_matrix_full_banded<trace_t>{};
+            else
+                return alignment_trace_matrix_full<trace_t>{};
         }
 
     public:
         //!\brief The matrix policy based on the configurations given by `config_type`.
-        using type = decltype(select<config_type>());
+        using type = deferred_crtp_base<alignment_matrix_policy,
+                                        decltype(select_score_matrix()),
```

Comment:
is abusing static function return types really nicer than a `std::conditional<>`?
Are you using the selection somewhere else?

Reply:
Yes, old stuff.